### PR TITLE
[TIMOB-23388] [TIMOB-24386] autoLink URL for Ti.UI.Label

### PR DESCRIPTION
**Pull request by infosia**
- [TIMOB-23388](https://jira.appcelerator.org/browse/TIMOB-23388) autoLink URL for Ti.UI.Label
- [TIMOB-24386](https://jira.appcelerator.org/browse/TIMOB-24386) Resize Label when root Window is resized

Test code: Following label should display hyperlink for each `http://` `tel:` `mailto:` schemes. Also, this label should be resized when you resize root Window (Windows Store app only).

```js
var win = Ti.UI.createWindow({ backgroundColor: 'green'});

var label1 = Ti.UI.createLabel({
    text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. http://www.appcelerator.com/ Ut enim ad tel:0123456789 minim veniam, quis nostrud mailto:foo@example.com exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
    autoLink: Titanium.UI.AUTOLINK_URLS,
    backgroundColor: 'white',
    width: Ti.UI.FILL, height: Ti.UI.SIZE
});

win.add(label1);
win.open();
```

### CODE REVIEW

**Comment by infosia** [comment]
test this please

File: Source/UI/src/Label.cpp [hunk 1]
```diff
@@ -93,15 +96,60 @@ namespace TitaniumWindows
...
+				std::regex::extended
+			);
+			std::list<std::string> blocks;
+			boost::split(blocks, get_text(), boost::is_any_of(" "));
+
+			std::smatch url_matches;
+			BOOST_FOREACH(std::string text, blocks) {
+
+				const auto run = ref new Run();
+				run->Text = TitaniumWindows::Utility::ConvertString(text);
```
Comment (garymathews) [inline Source/UI/src/Label.cpp]: `ConvertUTF8String()`

File: Source/UI/src/Label.cpp [hunk 2]
```diff
@@ -93,15 +96,60 @@ namespace TitaniumWindows
...
+
+				if ((enableALL || enableURL) && std::regex_search(text, url_matches, url_pattern) && !url_matches.str(1).empty()) {
+					auto hyperlink = ref new Hyperlink();
+					hyperlink->NavigateUri = ref new Windows::Foundation::Uri(TitaniumWindows::Utility::ConvertString(text));
```
Comment (garymathews) [inline Source/UI/src/Label.cpp]: `ConvertUTF8String()`

File: Source/UI/src/Window.cpp [hunk 3]
```diff
@@ -54,6 +54,17 @@ namespace TitaniumWindows
 				// Update back button visibility. Note that back button is available on Store App too.
 				SystemNavigationManager::GetForCurrentView()->AppViewBackButtonVisibility = AppViewBackButtonVisibility::Visible;
 			});
+			rootFrame->SizeChanged += ref new Windows::UI::Xaml::SizeChangedEventHandler([this](Platform::Object^, Windows::UI::Xaml::SizeChangedEventArgs^) {
```
Comment (garymathews) [inline Source/UI/src/Window.cpp]: This works, but for `Label` the text is limited by [label__->MaxWidth](https://github.com/appcelerator/titanium_mobile_windows/blob/master/Source/UI/src/Label.cpp#L57)

**Comment by garymathews** [comment]
CR: **PASS**
FT: **PASS**

Awesome! 👍

**Pull request merged**